test/verify/machineslib.py

```diff
@@ -1104,14 +1131,41 @@ def installWithErrorTest(dialog):
                                          start_vm=False,
                                          connection='session'))
         # try to INSTALL WITH ERROR
-        installWithErrorTest(TestMachines.VmDialog(self, "subVmTestCreate15", is_filesystem_location=True,
+        installWithErrorTest(TestMachines.VmDialog(self, "subVmTestCreate15", sourceType='file',
                                                    location=config.NOVELL_MOCKUP_ISO_PATH,
                                                    memory_size=900, memory_size_unit='GiB',
                                                    storage_size=10, storage_size_unit='MiB',
                                                    os_vendor=config.APPLE_VENDOR,
                                                    os_name=config.MACOS_X_LEOPARD,
                                                    connection='session'))
 
+        # test PXE Source
+        if self.provider == "libvirt-dbus":
+            self.machine.execute("virsh net-destroy default && virsh net-undefine default")
+
+            # Disable selinux because it makes TFTP directory inaccesible and we don't want sophisticated configuration for tests
+            self.machine.execute("setenforce 0")
+
+            # Set up the PXE server configuration files
+            self.machine.execute("mkdir -p /home/root/pxe-config && echo \"{0}\" > /home/root/pxe-config/pxe.cfg".format(PXE_SERVER_CFG))
+
+            # Define and start a NAT network with tftp server configuration
+            self.machine.execute("echo \"{0}\" > /tmp/pxe-nat.xml && virsh net-define /tmp/pxe-nat.xml && virsh net-start {1}".format(NETWORK_XML_PXE, 'pxe-nat'))
+
+            # We don't handle events for networks yet, so reload the page to refresh the state
+            self.browser.reload()
+            self.browser.enter_page('/machines')
+            self.browser.wait_in_text("body", "Virtual Machines")
+
+            runner.tryCreate(TestMachines.VmDialog(self, "subVmTestCreate16", sourceType='pxe',
+                                                   location="Virtual Network pxe-nat: NAT",
+                                                   memory_size=1, memory_size_unit='GiB',
+                                                   storage_size=100, storage_size_unit='MiB',
+                                                   os_vendor=config.NOVELL_VENDOR,
+                                                   os_name=config.NOVELL_NETWARE_6,
+                                                   start_vm=True))
+            self.browser.wait_in_text("#vm-subVmTestCreate16-state", "running")
```

Comment:
Do we have a way to validate that the PXE boot actually worked? "Running" just means the (virt-)phsyical machine state, so it could just hang in the BIOS/EFI with no working boot device?

Reply:
I was struggling with this to be honest. I am aware that it's not 'fully' tested like this. I will try to find a more fine grained way to test this.